py/server/tests/test_partitioned_table_service.py

```diff
@@ -0,0 +1,248 @@
+                datetime_col(name="Datetime", data=[1, -1]),
+            ]
+            yield dharrow.to_arrow(new_table(cols=cols))
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.pa_table = next(self.gen_pa_table())
+        self.test_table = dharrow.to_table(self.pa_table)
+
+    def test_make_table_without_partition_schema(self):
+        backend = TestBackend(self.gen_pa_table(), pt_schema=self.pa_table.schema)
+        data_service = PythonTableDataService(backend)
+        table = data_service.make_table(TableKey("test"), live=False)
+        self.assertIsNotNone(table)
+        self.assertEqual(table.columns, self.test_table.columns)
+        table = None # what happens when table is GC'd? LivenessScope will release it?
+
+    def test_make_static_table_with_partition_schema(self):
+        pc_schema = pa.schema(
+            [pa.field(name="Ticker", type=pa.string()), pa.field(name="Exchange", type=pa.string())])
+        backend = TestBackend(self.gen_pa_table(), pt_schema=self.pa_table.schema, pc_schema=pc_schema)
+        data_service = PythonTableDataService(backend)
+        table = data_service.make_table(TableKey("test"), live=False)
+        self.assertIsNotNone(table)
+        self.assertTrue(table.columns[0].column_type == ColumnType.PARTITIONING)
+        self.assertTrue(table.columns[1].column_type == ColumnType.PARTITIONING)
+        self.assertEqual(table.columns[2:], self.test_table.columns[2:])
+        self.assertEqual(table.size, 2)
+        self.assertEqual(backend.existing_partitions_called, 1)
+        self.assertEqual(backend.partition_size_called, 1)
+        # how is the table different from the PartitionedTable?
+
+    def test_make_live_table_with_partition_schema(self):
+        pc_schema = pa.schema(
+            [pa.field(name="Ticker", type=pa.string()), pa.field(name="Exchange", type=pa.string())])
+        backend = TestBackend(self.gen_pa_table(), pt_schema=self.pa_table.schema, pc_schema=pc_schema)
+        data_service = PythonTableDataService(backend)
+        table = data_service.make_table(TableKey("test"), live=True)
+        self.assertIsNotNone(table)
+        self.assertTrue(table.columns[0].column_type == ColumnType.PARTITIONING)
+        self.assertTrue(table.columns[1].column_type == ColumnType.PARTITIONING)
+        self.assertEqual(table.columns[2:], self.test_table.columns[2:])
+        self.wait_ticking_table_update(table, 20, 5)
+        self.assertGreaterEqual(table.size, 20)
+        self.assertEqual(backend.existing_partitions_called, 0)
+        self.assertEqual(backend.partition_size_called, 0)
+
+    def test_make_live_table_with_partition_schema_ops(self):
+        pc_schema = pa.schema(
+            [pa.field(name="Ticker", type=pa.string()), pa.field(name="Exchange", type=pa.string())])
+        backend = TestBackend(self.gen_pa_table(), pt_schema=self.pa_table.schema, pc_schema=pc_schema)
+        data_service = PythonTableDataService(backend)
```

Comment:
I really don't love this class name.  I already mentioned `Python`.  Then we take 3 words that we use all over the code, with all sorts of meanings, and cram them together.  Can we get a more clear and more unique name?

Reply:
I'm fine with dropping "Python" from the name, although that just makes my IDE searching more annoying, as has already happened for "table". `TableDataService` is what the underlying infrastructure is called, and has been for more than a decade.